app/src/main/java/org/mozilla/focus/session/IntentProcessor.kt

```diff
@@ -84,19 +84,19 @@ class IntentProcessor(
                         // HomeScreen.BLOCKING_ENABLED
 
                         createSession(
-                            SessionState.Source.HOME_SCREEN,
+                            SessionState.Source.Internal.HomeScreen,
                             intent,
                             intent.dataString ?: "",
                             requestDesktop
                         )
                     }
                     intent.hasExtra(TextActionActivity.EXTRA_TEXT_SELECTION) -> createSession(
-                        SessionState.Source.TEXT_SELECTION,
+                        SessionState.Source.Internal.TextSelection,
                         intent,
                         intent.dataString ?: ""
                     )
                     else -> createSession(
-                        SessionState.Source.ACTION_VIEW,
+                        SessionState.Source.External.ActionView(null),
```

Comment:
@grigoryk can you assist on what the proper values we should be passing for the external sources? I am also curious if `null` is acceptable here for Focus.

Reply:
`null` is fine here, yeah. Basically we'll just be omitting information about the refferrerring app, which is not a problem if the application isn't going to use it. This is also possible in Fenix if it's running on API21.

If you did want to integrate external referrer, you can get at the information about package/category at the activity level: https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt#L106-L127